doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+associated args. For some of these actions, with very few arguments, the strings
+can be relatively simple. Take for example, `CopyText`:
+
+JSON | Generated String
+-- | --
+`{ "action":"copyText" }` | "Copy text"
+`{ "action":"copyText", "singleLine": true }` | "Copy text as a single line"
+`{ "action":"copyText", "singleLine": false, "copyFormatting": false }` | "Copy text without formatting"
+`{ "action":"copyText", "singleLine": true, "copyFormatting": true }` | "Copy text as a single line without formatting"
+
+CopyText is a bit of a simplistic case however, with very few args or
+permutations of argument values. For things like `newTab`, `splitPane`, where
+there are many possible arguments and values, it will be acceptable to simply
+append `", property:value"` strings to the generated names for each of the set
+values.
+
+For example:
+
+JSON | Generated String
+-- | --
+`{ "action":"newTab", "profile": "Hello" }` | "Open a new tab, profile:Hello"
+`{ "action":"newTab", "profile": "Hello", "directory":"C:\\", "commandline": "wsl.exe", title": "Foo" }` | "Open a new tab, profile:Hello, directory:C:\\, commandline:wsl.exe, title:Foo"
+
+
+This is being chosen in favor of something that might be more human-friendly,
+like "Open a new tab with profile {profile name} in {directory} with
+{commandline} and a title of {title}". This string would be much harder to
+synthesize, especially considering localization concerns.
+
+#### Remove the resource key notation
+
+Since we'll be using localized names for each of the actions in `GenerateName`,
+we no longer _need_ to provide the `{ "name":{ "key": "SomeResourceKey" } }`
+syntax introduced in the original spec. This functionality was used to allow us
+to define localizable names for the default commands.
+
+However, I think we should keep this functionality, to allow us additional
+flexibility when defining default commands.
+
+### Complete Defaults
+
+Considering both of the above proposals, the default keybindings and commands
+will be defined as follows:
+
+* The current default keybindings will be untouched. These actions will
+  automatically be added to the Command Palette, using their names generated
+  from `GenerateName`.
+  - **TODO: FOR DISCUSSION**: Should we manually set the names for the default
+    "New Tab, profile index: 0" keybindings to `null`? This seems like a not
+    terribly helpful name for the Command Palette, especially considering the
+    iterable commands listed below.
+* We'll add a few new commands:
+  - A nested, iterable command for "Open new tab with
+    profile..."/"Profile:{profile name}"
+  - A nested, iterable command for "Select color scheme..."/"{scheme name}"
+  - A nested, iterable command for "New Pane..."/"Profile:{profile
+    name}..."/["Automatic", "Horizontal", "Vertical"]
+  > 👉 NOTE: These default nested commands can be removed by the user defining
+  > `{ "name": "Open new tab with profile...", "action":null }` (et al) in their

```

Comment:
I am not fully comfortable with `name` being the key of repute here. If we use resource keys for these (which we **should**!). It's not a durable identifier, **ESPECIALLY** if the localizers come back and refine the translation in the future.

Reply:
(The localizations change sometimes!)

Reply:
VS has this mapping:

![image](https://user-images.githubusercontent.com/189190/85079108-de2f7500-b17a-11ea-96a7-02104e7da678.png)

![image](https://user-images.githubusercontent.com/189190/85079166-fbfcda00-b17a-11ea-98b2-4f99105086c7.png)


I'm guessing they map...

key -> id

menu -> id